### PR TITLE
[Reviewer: Ellie] Exception handling: health-checker code

### PR DESCRIPTION
**Pull request by rkd-msw**
This integrates https://github.com/Metaswitch/cpp-common/pull/241 with Sprout. As we agreed, I've just made any INVITE/200 pass health-checking for simplicity; I haven't done anything to ignore I-CSCF 200 OKs if we have an S-CSCF. I've updated my copy of the spec to say so - I'll send that to you for a quick review once I've implemented the Homestead/Chronos/Ralf code, in case implementing those leads to any spec changes.

I've tested:
* pretty extensively in UT (see below)
* live, by checking for the `19-02-2015 17:08:06.773 UTC Debug health_checker.cpp:68: Check for overall system health passed - not aborting` log in Sprout

### CODE REVIEW

**Comment by rkd-msw** [comment]
I've switched to a condition variable, so this is now ready for review. I've tested the new code:
* with the same UTs
* by running live and checking that a log is generated once a minute
* by terminating under valgrind and checking that nothing was leaked - but I couldn't confirm this, because I hit https://github.com/Metaswitch/sprout/issues/853 rather than exiting cleanly.

File: sprout/main.cpp [hunk 1]
```diff
@@ -1716,11 +1716,17 @@ int main(int argc, char* argv[])
   Counter* overload_counter =
       new StatisticCounter("rejected_overload",
                            stack_data.stats_aggregator);
+  HealthChecker* health_checker = new HealthChecker();
+  pthread_t health_check_thread;
+  pthread_attr_t health_check_attr;
+  pthread_attr_init(&health_check_attr);
```
Comment (eleanor-merry) [inline sprout/main.cpp]: Does this attr get used at all? Also can you change the line length below

File: sprout/ut/common_sip_processing_test.cpp [hunk 2]
```diff
@@ -366,3 +442,108 @@ TEST_F(CommonProcessingTest, BadResponseDropped)
   // Expect it to just vanish.
   ASSERT_EQ(0, txdata_count());
 }
+
+// If:
+//  - an exception has already been hit
+//  - the health-checker runs a check
+//  - and all the requests seen recently have not had a response
+// then Sprout should be aborted.
+TEST_F(CommonProcessingTest, DeathTest_MissingResponseFailsHealthCheck)
```
Comment (eleanor-merry) [inline sprout/ut/common_sip_processing_test.cpp]: As discussed, can you tidy up the warnings here?

**Comment by eleanor-merry** [comment]
Reviewed

**Comment by rkd-msw** [comment]
@eleanor-merry , I've applied your markups.

**Pull request closed without merging**